src/common/store/synchronization.middleware.ts

```diff
@@ -1,56 +1,100 @@
 import type { Middleware } from '@reduxjs/toolkit';
 import { PERSIST } from 'redux-persist';
 
+import { init, sign } from '@common/store/signing.slice';
 import synchronization, {
   getHandshaken,
   getTargetPublicKey,
-  sendPublicKey
+  sendPublicKey,
+  SynchronizationTarget
 } from '@common/store/synchronization.slice';
 import { encryptJson } from '@common/utils';
 import type { ReduxIPC } from '@types';
 
+import { saveAccountSecrets } from './accounts.slice';
 import { setPersistor } from './persistence.slice';
 
 /**
  * An array of action paths that will not be synchronised with the other process.
  */
 export const IGNORED_PATHS = [synchronization.name];
 export const IGNORED_ACTIONS = [PERSIST, setPersistor.type];
+export const SIGNING_ACTIONS = [init.type, sign.type, saveAccountSecrets.type];
 
 /**
  * Middleware that dispatches any actions to the other Electron process.
  * @param ipc The Electron process to dispatch from.
  */
-export const synchronizationMiddleware = (ipc: ReduxIPC): Middleware => (store) => (next) => (
-  action
-) => {
+export const synchronizationMiddleware = (
+  ipcs: Partial<Record<SynchronizationTarget, ReduxIPC>>,
+  self: SynchronizationTarget
+): Middleware => (store) => (next) => (action) => {
   const path = action.type.split('/')[0];
+
+  // SIGNING and RENDERER only communicates with MAIN
+  // MAIN communicates with both, mostly relaying requests from RENDERER
+  const ipcMapping = {
+    [SynchronizationTarget.SIGNING]: SynchronizationTarget.MAIN,
+    [SynchronizationTarget.MAIN]:
+      action.from === SynchronizationTarget.RENDERER || SIGNING_ACTIONS.includes(action.type)
+        ? SynchronizationTarget.SIGNING
+        : SynchronizationTarget.RENDERER,
+    [SynchronizationTarget.RENDERER]: SynchronizationTarget.MAIN
+  };
+
+  const encryptionMapping = {
+    ...ipcMapping,
+    [SynchronizationTarget.RENDERER]: SIGNING_ACTIONS.includes(action.type)
+      ? SynchronizationTarget.SIGNING
+      : SynchronizationTarget.MAIN
+  };
+
+  const target = ipcMapping[self];
+
+  const encryptionTarget = encryptionMapping[self];
+
+  // UNDEFINED = ALL
+  const to = action.to ?? SIGNING_ACTIONS.includes(action.type) ? encryptionTarget : undefined;
```

Comment:
Maybe add an `All` value to the enum. That´s a bit more clear than using `undefined` here.

Reply:
I did this initially, but then it would require the mappings above to include `All` to not be a typing mess. But that doesn't really make sense